internal/cmd/egctl/translate.go

```diff
@@ -389,6 +391,13 @@ func kubernetesYAMLToResources(str string) (string, *gatewayapi.Resources, error
 		un := unstructured.Unstructured{Object: obj}
 		gvk := un.GroupVersionKind()
 		name, namespace := un.GetName(), un.GetNamespace()
+		if namespace == "" {
```

Comment:
so the Gateway API translator doesnt like the fact that the resource `Namespace` field is unset or that the `default` Namespace resource is not part of the input ?

Reply:
another thought, rather than automagaically inserting resources by default, how about adding a flag called `--add-missing-resources` which explicitly does this on behalf on the user

Reply:
This PR intends to emulate the behavior of kubectl, which uses the current namespace if the resources in the applied yaml don't have one.
With this change, a bundle of resources used by the kubectl to setup the environment (here the quickstart.yaml in this project) can also be accepted by the egctl.

As for `--add-missing-resources`, IMO maybe we can make it the default. Therefore the user can specify a single HTTPRoute and let egctl provide a default gateway/gatewayclass/service and so on. It would be very user-friendly if we can get rid of the boilerplate. What about discussing it in another issue?

Reply:
sure sgtm

Reply:
raised https://github.com/envoyproxy/gateway/issues/1122